app/controllers/api/actions_controller.rb

```diff
@@ -0,0 +1,59 @@
+class Api::ActionsController < Api::ApiController
+
+  def create
+    payload = JSON.parse(params[:payload])
+    user = User.find_by_slack_id(payload['user']['id'])
+
+    if user.nil?
+      respond_with_error
+      return
+    end
+
+    callback_id = payload['callback_id']
+    action_name = payload['actions'].first['name']
+    action_value = payload['actions'].first['value']
+
+    case callback_id
+    when 'tasks'
+      case action_name
+      when 'complete'
```

Comment:
Yeah, definitely. I actually had it that way originally but I decided to just leave it so I didn't have to worry about passing vars around. I figured once this method got far too long the abstraction would happen naturally.